agent/engine/dependencygraph/graph.go

```diff
@@ -405,8 +410,12 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 
 	case successCondition:
 		// The 'target' container desires to be moved to 'Created' or the 'steady' state.
-		// Allow this only if the known status of the dependency container state is stopped with an exit code of 0
-		if dependsOnContainer.GetKnownExitCode() != nil {
+		// Allow this only if the environment variable 'ECS_PULL_PARALLEL_DEPENDSON_SUCCESS' is set to true and the
+		// target container known status has not reached to 'Pulled'; otherwise, the known status of the dependency
+		// container state is stopped with an exit code of 0 if the environment variable is false as default
+		if cfg.ContainerPullInParallel.Enabled() && targetContainerKnownStatus < apicontainerstatus.ContainerPulled {
```

Comment:
Do u know why is this condition being checked only for success condition? Should it not happen for all the other conditions?

Reply:
there hasn't been a request for other dependsOn condition. we are checking the current pull behavior for other conditions. if things don't break, we can extend this for other conditions as well.

Reply:
The environment variable is renamed to ` ECS_PULL_DEPENDENT_CONTAINER_PARALLEL ` and can be applied to `dependsOn condition: START/SUCCESS/COMPLETE/HEALTHY`. Changes are made in [this commit](https://github.com/aws/amazon-ecs-agent/pull/2700/commits/fba2d68cb35ed883fb1b920417c02caa77a43981#diff-9a13130c1043999591972f25c335c69b979715c5b45009b5d618c043d709a45aR394-R395). Thank you!